src/exporter.js

```diff
@@ -87,13 +79,16 @@ function exporter (hash, dagService, options, callback) {
           return
         })
       }
-      ee.emit('file', { stream: rs, path: name, dir: dir })
+      this.push({ stream: rs, path: name, dir: dir })
+      if (callback) {
+        callback()
+      }
+      return
     }
   }
 
-  function dirExporter (node, name, callback) {
+  this.dirExporter = (node, name, callback) => {
```

Comment:
Do these methods need to be on `this`? If possible just use `function` -- it keeps the methods unexposed to the outside world.